cg/apps/lims/api.py

```diff
@@ -415,22 +415,28 @@ def _get_all_step_dates(self, step_names_udfs, lims_id, artifact_type=None):
         and optionally the type
         """
         dates = []
-
+        all_artifacts = []
+        processes = []
         for process_type in step_names_udfs:
             artifacts = self.get_artifacts(
                 process_type=process_type, samplelimsid=lims_id, type=artifact_type
             )
 
             for artifact in artifacts:
                 udf_key = step_names_udfs[process_type]
-                if artifact.parent_process and artifact.parent_process.udf.get(udf_key):
+                parent_process = artifact.parent_process
+                if parent_process and parent_process.udf.get(udf_key):
                     dates.append(
-                        (
-                            artifact.parent_process.date_run,
-                            artifact.parent_process.udf.get(udf_key),
-                        )
+                        (parent_process.date_run, parent_process.udf.get(udf_key),)
                     )
-
+                processes.append(parent_process.id)
+                all_artifacts.append(artifact)
+
+        if all_artifacts and not dates:
+            udfs = " ,".join(set(step_names_udfs.values()))
+            process_ids = " ,".join(set(processes))
+            msg = f"Did not find date for sample {lims_id}. Searced for udfs: {udfs}. Searched in processes: {process_ids}."
+            log.error(msg)
```

Comment:
```suggestion
```

This is not an error but the expected behavior until someone enters a date in LIMS as I understand it

Reply:
Yes. True.  Should it be log.warning instead?

Reply:
maybe I'm wrong. If there is an artifact then there must be a date?

Reply:
ok, lets warn only since it could be a legitimate state